### PR TITLE
Fix for "Keep the remote version" deletes files on new development theme

### LINKED ISSUE #2433: [Bug]: shopify --theme-serve "Keep the remote version" deletes files on new development theme
### Please confirm that you have:

- [X] Searched [existing issues](https://github.com/Shopify/shopify-cli/issues) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://github.com/Shopify/shopify-cli/releases).

### What type of project are you working on?

theme

### Expected behavior

When doing "shopify theme serve" anew (No development theme uploaded yet) it says that all .json files are out of sync. "is different from the remote version in the development theme", choosing "Keep the remote version" actually results in all of your .json files being deleted! 

It shouldn't say any files are out of sync because they don't yet exist on the development theme.. 

### Actual behavior

All of my .json files were deleted

### Stack trace

_No response_

### Reproduction steps

1. Login to a Shopify Store. (shopify login --store [store-a.myshopify.com](http://store-a.myshopify.com/))
2. Login to a different Shopify Store ((shopify login --store [store-b.myshopify.com](http://store-b.myshopify.com/)))
3. Run "shopify theme serve --theme-editor-sync"
4. Wait for message "xx is different from the remote version in the development theme"
5. Choose number 1 (Keep the remote version)
6. Watch as all of your .json files get deleted! 

### Operating System

Windows 11

### CLI version (run `shopify version` if you're not sure)

2.19.0

### Shell

Git Bash

### Ruby version (run `ruby -v` if you're not sure)

2.7.3p183

### What language and version are you using in your application?

N/A

### PR DESCRIPTION
**Pull request by mgmanzella**
### WHY are these changes introduced?

Fixes #2433 

### WHAT is this pull request doing?

- Adds `@created_at_runtime` variable that gets set to true if development theme is created
- Defaults to using local files `@created_at_runtime=true`

#### New development themes defaults to local files
- In your partners dashboard, create a new dev store
- Create a new staff account for your work email if your partners account isn't associated with your work email (you need to do this so you can you use the cli with the new store)
- Accept invite for staff account 

##### In a new terminal

```
$ shopify theme init
$ cd <name of theme>
$ git init
$ git add —all
$ git commit -am "init"
$ pwd
```
Copy the path printed after `pwd`

##### In another terminal 
```
$ git fetch
$ git checkout fix-remote-file-local-delete
$ shopify-dev login —store <new store name>
$ shopify-dev theme serve </Path/to/theme/ that you copied> --theme-editor-sync
```

Confirm local files get uploaded and don't get deleted. Some files do get updated locally though

#### Existing development theme prompts user for strategy

In the theme directory run `git reset --hard` to restore the files. In the shopify-cli directory run `shopify-dev theme serve </Path/to/theme/> --theme-editor-sync` again and confirm you're prompted to pick a sync strategy

### Post-release steps
None

### Update checklist

- [x] I've added a CHANGELOG entry for this PR (if the change is public-facing)
- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows).
- [x] I've left the version number as is (we'll handle incrementing this when releasing).
- [x] I've included any post-release steps in the section above (if needed).

### CODE REVIEW

File: lib/shopify_cli/theme/syncer.rb [hunk 1]
```diff
@@ -387,7 +387,11 @@ def backoff_if_near_limit!(used, limit)
       end
 
       def overwrite_json?
-        @overwrite_json
+        theme_created_at_runtime? | @overwrite_json
```
Comment (karreiro) [inline lib/shopify_cli/theme/syncer.rb:390]: Do we have any specific reason to use the binary `|` instead of the logical `||`?

**Review by karreiro — APPROVED** [review]
Thank you, @mgmanzella! Elegant PR 🚀 

I've left one minor/optional comment to clarify a detail, but the PR looks good and works fine as well 🎩

Thanks!

File: lib/shopify_cli/theme/syncer.rb (hunk 1 above)
Comment (mgmanzella) [reply lib/shopify_cli/theme/syncer.rb:390]: nope! just got my ruby syntax mixed up 😅 ill change it to the logical operator so we don't confuse ourselves when we come back to the code loool

File: lib/shopify_cli/theme/development_theme.rb [hunk 2]
```diff
@@ -9,6 +9,8 @@ module Theme
     API_NAME_LIMIT = 50
 
     class DevelopmentTheme < Theme
+      attr_accessor :created_at_runtime
```
Comment (jessetuglu) [inline lib/shopify_cli/theme/development_theme.rb:12]: Curious – since we have a wrapper method `created_at_runtime?`, is this accessor needed?

**Review by ulgut — APPROVED** [review]

**Pull request merged**